parachain/runtime/src/lib.rs

```diff
@@ -771,13 +794,62 @@ parameter_types! {
     pub const InflationPeriod: BlockNumber = YEARS;
 }
 
+pub struct DealWithRewards;
+
+impl supply::OnInflation<AccountId> for DealWithRewards {
+    type Currency = NativeCurrency;
+    fn on_inflation(from: &AccountId, amount: Balance) {
+        // transfer will only fail if balance is too low
+        // existential deposit is not required due to whitelist
+        let _ = Self::Currency::transfer(
+            from,
+            &AnnuityAccount::get(),
+            token_distribution::VAULT_INFLATION_REWARDS * amount,
```

Comment:
can we use checked math?

Reply:
Not for this because `Permill` doesn't implement the necessary traits.

Reply:
Already safe https://github.com/paritytech/substrate/blob/0e6cc5668d9ee8d852a3aa3f85a2ab5fcb4c75a1/primitives/arithmetic/src/per_things.rs#L753-L763